cirq-ft/cirq_ft/algos/qrom.py

```diff
@@ -166,14 +166,10 @@ def decompose_zero_selection(
             context.qubit_manager.qfree(and_ancilla + [and_target])
 
     def _break_early(self, selection_index_prefix: Tuple[int, ...], l: int, r: int):
-        global_unique_element: Set[int] = set()
         for data in self.data:
             unique_element = np.unique(data[selection_index_prefix][l:r])
             if len(unique_element) > 1:
                 return False
-            global_unique_element.add(unique_element[0])
```

Comment:
In the original PR, my idea was to do the optimization only when the element to be loaded is _identical_ across all data sequences. This is an overkill since all we really need is that within each data sequence, we have only one element to load. That specific element can be different across data sequences. The set was checking the uniqueness condition across datasets which was redundant.